From 842b4b5d995f23dfe404f550bd4df73ff304270a Mon Sep 17 00:00:00 2001
From: ShinDarth <borzifrancesco@gmail.com>
Date: Fri, 8 Aug 2014 12:50:40 +0200
Subject: [PATCH] PvPstats system

Signed-off-by: ShinDarth <borzifrancesco@gmail.com>
---
 src/game/BattleGround/BattleGround.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/game/BattleGround/BattleGround.cpp b/src/game/BattleGround/BattleGround.cpp
index 80cfbd4..8cfbd3b 100644
--- a/src/game/BattleGround/BattleGround.cpp
+++ b/src/game/BattleGround/BattleGround.cpp
@@ -629,17 +629,25 @@ void BattleGround::EndBattleGround(Team winner)
     WorldPacket data;
     int32 winmsg_id = 0;
 
+    int32 level = GetMaxLevel() / 10;
+
     if (winner == ALLIANCE)
     {
         winmsg_id = isBattleGround() ? LANG_BG_A_WINS : LANG_ARENA_GOLD_WINS;
 
         PlaySoundToAll(SOUND_ALLIANCE_WINS);                // alliance wins sound
+
+        if (isBattleGround())
+            CharacterDatabase.PQuery("INSERT INTO pvpstats_faction (faction, level, date) VALUES (0, %d, NOW());", level);
     }
     else if (winner == HORDE)
     {
         winmsg_id = isBattleGround() ? LANG_BG_H_WINS : LANG_ARENA_GREEN_WINS;
 
         PlaySoundToAll(SOUND_HORDE_WINS);                   // horde wins sound
+
+        if (isBattleGround())
+            CharacterDatabase.PQuery("INSERT INTO pvpstats_faction (faction, level, date) VALUES (1, %d, NOW());", level);
     }
 
     SetWinner(winner);
@@ -745,6 +753,9 @@ void BattleGround::EndBattleGround(Team winner)
             RewardMark(plr, ITEM_WINNER_COUNT);
             RewardQuestComplete(plr);
             plr->GetAchievementMgr().UpdateAchievementCriteria(ACHIEVEMENT_CRITERIA_TYPE_WIN_BG, 1);
+
+            if (isBattleGround())
+                CharacterDatabase.PQuery("INSERT INTO pvpstats_players (character_guid, level, date) VALUES (%d, %d, NOW());", plr->GetUInt64Value(0), level);
         }
         else
             RewardMark(plr, ITEM_LOSER_COUNT);
-- 
1.9.1

